src/sssom/util.py

```diff
@@ -1126,21 +1125,10 @@ def curie_from_uri(uri: str, prefix_map: Mapping[str, str]) -> str:
     >>> curie_from_uri("hgnc.genegroup:1234", {})
     'hgnc.genegroup:1234'
     """
-    # TODO consider replacing with :func:`bioregistry.curie_from_iri`
-    # FIXME what if the curie has a subspace in it? RE will fail
-    if is_curie(uri):
-        return uri
-    for prefix in prefix_map:
-        uri_prefix = prefix_map[prefix]
-        if uri.startswith(uri_prefix):
-            remainder = uri.replace(uri_prefix, "")
-            curie = f"{prefix}:{remainder}"
-            if is_curie(curie):
-                return f"{prefix}:{remainder}"
-            else:
-                logging.warning(f"{prefix}:{remainder} is not a CURIE ... skipping")
-                continue
-    raise NoCURIEException(f"{uri} does not follow any known prefixes")
+    warnings.warn("Use safe_compress() instead", DeprecationWarning)
+    if not isinstance(prefix_map, Converter):
+        converter = Converter.from_prefix_map(prefix_map)
+    return safe_compress(uri, converter)
```

Comment:
What happens if `prefix_map` _is_ an instance of `converter`? I mean, this case should basically never happen, but shouldn't there be an

```
else: converter = prefix_map 
```

There?

Reply:
oop this is a typo, thank you for the catch.

Reply:
Fixed in ec3452f